RiotSwiftUI/Modules/Settings/Notifications/Service/NotificationSettingsServiceType.swift

```diff
@@ -40,5 +40,16 @@ protocol NotificationSettingsServiceType {
     ///   - ruleId: The id of the rule.
     ///   - enabled: Whether the rule should be enabled or disabled.
     ///   - actions: The actions to update with.
-    func updatePushRuleActions(for ruleId: String, enabled: Bool, actions: NotificationActions?)
+    ///   - completion: The completion of the operation.
+    func updatePushRuleActions(for ruleId: String, enabled: Bool, actions: NotificationActions?, completion: ((Result<Void, Error>) -> Void)?)
```

Comment:
As a speculative thought what would happen if we made this method `async throws` and having the
 `completion` variant (which is used less often and only with obj-c) the extension if even necessary? I think it would make the code more readable.

Reply:
I think it should work as well.
I preferred to keep the non async version in the first place to minimize the refactor I needed to do.
But at this point it should be an easy one.

Reply:
I think it may also help with some of the `Result<Void, Error>` workarounds and magic

Reply:
I refactored several things trying to leverage more async/await to make the code easier to be read.
I also found some already existing memory leak and I fixed that as well.
https://github.com/vector-im/element-ios/pull/7320/commits/6850633a52e6a5aa536afc803a8e144ef577a3e9